api/v1beta1/types_template.go

```diff
@@ -86,6 +86,8 @@ type VnetTemplateSpec struct {
 }
 
 // VnetPeeringsTemplateSpec defines a list of peerings of the newly created virtual network with existing virtual networks.
+// +listType=map
+// +listMapKey=remoteVnetName
```

Comment:
same question here, I don't know if remoteVnetName is a unique identifier

Reply:
it's the only property in the underlying struct inside the slice, is it possible we'd be retaining more than one items in this slice that are identical?

Reply:
Could it be that atomic merge behavior is good enough for this field, as there are no two clients trying to set different parts of the slice?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b59c9198f45c3cb13a0b1209cb8167ebcc4156ac/api/v1beta1/azurecluster_validation.go#L266

uniqueness comes from resource group + remote vnet name for a vnet peering, this again the same problem as below where VnetPeeringsTemplateSpec is only a subset of VnetPeeringsSpec and is missing resource group, which might be an issue if you want to specify more than one vnet with the same name (which would be a valid thing to do if you have two vnets in two different RG with the same name).

Reply:
Sounds like we really don't have a unique identifier in `VnetPeeringClassSpec` and in this case we shouldn't introduce resourceGroup to get to a map key consisting of resourceGroup and remoteVnetName. (as resourceGroup should be different per cluster)

Given:
* that per default we have atomic merge behavior and 
* we only need a map key to allow the `map` merging behavior ("At any point in time, a single manager owns the list. If set or map, different managers can manage entries separately.")

Do we have a use caseI were we want different managers (controllers) to be able to manage/own different entries of this list?